core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -605,6 +606,66 @@ private[spark] object Utils extends Logging {
     }
   }
 
+  def unZip(inFile: File, unzipDir: File): Unit = {
```

Comment:
Can we refer to other references of codes? e.g., https://cs.android.com/android/platform/superproject/+/master:tools/tradefederation/core/common_util/com/android/tradefed/util/ZipUtil2.java;drc=master;l=44

It has a bunch of corner cases handling. .e.g., we will have to skip this logic if the platform is non-Unix.

Reply:
Yes, your suggestion is right, we really can not cover all scenarios, I refer to, for the logic of granting file permissions added judgment, for non-UNIX platforms, skip this logic.